srcs/parser/parser_utils.c

```diff
@@ -0,0 +1,117 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parser_utils.c                                     :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: jteissie <jteissie@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/08/16 18:12:37 by jteissie          #+#    #+#             */
+/*   Updated: 2024/08/16 18:46:27 by jteissie         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "parser.h"
+
+t_bool	is_invalid_char(char c)
+{
+	if (c == '1' || c == '0')
+		return (FALSE);
+	else if (c == ' ' || c == '\n')
+		return (FALSE);
+	else if (c == 'N' || c == 'S' || c == 'W' || c == 'E')
+		return (FALSE);
+	return (TRUE);
+}
+
+t_parse_status	check_invalid_chars(char **map)
+{
+	uint32_t	x;
+	uint32_t	y;
+
+	x = 0;
+	y = 0;
+	while (map[x])
+	{
+		while (map[x][y])
+		{
+			if (is_invalid_char(map[x][y]) == TRUE)
+				return (MAP_ERR);
+			y++;
+		}
+		x++;
+		y = 0;
+	}
+	return (MAP_OK);
+}
+
+void	fill_whitespaces(char **map)
+{
+	uint32_t	x;
+	uint32_t	y;
+
+	x = 0;
+	y = 0;
+	while (map[x])
+	{
+		while (map[x][y])
+		{
+			if (map[x][y] == ' ')
+				map[x][y] = '1';
+			y++;
+		}
+		y = 0;
+		x++;
+	}
+}
+
+static uint32_t	is_cardinal_pos(char c)
+{
+	return (c == NORTH || c == SOUTH || c == WEST || c == EAST);
+}
+
+t_parse_status	find_start(uint32_t coordinates[], char **map)
```

Comment:
Logic here is lacking. Starting position can never be on the edge of the map

Reply:
This is too annoying to check since I need to check for overflows and underflows, so 4 directions in total, let's just let the floodfill algorithm throw the exception instead of one-upping it.